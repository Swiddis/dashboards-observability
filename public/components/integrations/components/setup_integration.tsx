/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import * as Eui from '@elastic/eui';
import React, { useState, useEffect } from 'react';
import { string } from 'joi';
import { coreRefs } from '../../../framework/core_refs';

interface IntegrationConfig {
  displayName: string;
  connectionType: string;
  connectionDataSource: string;
}

interface IntegrationConfigProps {
  config: IntegrationConfig;
  updateConfig: (updates: Partial<IntegrationConfig>) => void;
}

const INTEGRATION_DATA_TABLE_COLUMNS = [
  {
    field: 'field',
    name: 'Field',
  },
  {
    field: 'sourceType',
    name: 'Source Data Type',
  },
  {
    field: 'destType',
    name: 'Destination Data Type',
  },
  {
    field: 'group',
    name: 'Mapping Group',
  },
];

// TODO support localization
const INTEGRATION_CONNECTION_DATA_SOURCE_TYPES: Map<string, string> = new Map([
  ['s3', 'table'],
  ['index', 'index'],
]);

const integrationConnectionSelectorItems = [
  {
    value: 's3',
    text: 'S3 Connection',
    dataSourceName: ['table', 'tables'],
  },
  {
    value: 'index',
    text: 'OpenSearch Index',
    dataSourceName: ['index', 'indexes'],
  },
];

const integrationDataTableData = [
  {
    field: 'domain.bytes',
    sourceType: 'string',
    destType: 'integer',
    group: 'communication',
  },
  {
    field: 'http.url',
    sourceType: 'string',
    destType: 'keyword',
    group: 'http',
  },
  {
    field: 'destination.address',
    sourceType: 'string',
    destType: 'keyword',
    group: 'communication',
  },
  {
    field: 'netflow.error_rate',
    sourceType: 'string',
    destType: 'string',
    group: 'http',
  },
  {
    field: 'http.route',
    sourceType: 'string',
    destType: 'string',
    group: 'http',
  },
  {
    field: 'destination.packets',
    sourceType: 'integer',
    destType: 'integer',
    group: 'communication',
  },
];

const suggestDataSources = async (type: string): Promise<Array<{ label: string }>> => {
  const http = coreRefs.http!;
  try {
    if (type === 'index') {
      const result = await http.post('/api/console/proxy', {
        body: '{}',
        query: {
          path: '_data_stream/ss4o_*',
          method: 'GET',
        },
      });
      return (
        result.data_streams?.map((item: { name: string }) => {
          return { label: item.name };
        }) ?? []
      );
    } else if (type === 's3') {
      const result = (await http.post('/api/console/proxy', {
        body: '{}',
        query: {
          path: '_plugins/_query/_datasources',
          method: 'GET',
        },
      })) as Array<{ name: string; connector: string }>;
      return (
        result
          ?.filter((item) => item.connector === 'S3GLUE')
          .map((item) => {
            return { label: item.name };
          }) ?? []
      );
    } else {
      console.error(`Unknown connection type: ${type}`);
      return [];
    }
  } catch (err: any) {
    console.error(err.message);
    return [];
  }
};

export function IntegrationDataModal({ close }: { close: () => void }): React.JSX.Element {
  return (
    <Eui.EuiModal onClose={close}>
      <Eui.EuiModalHeader>
        <h2>Data Table</h2>
      </Eui.EuiModalHeader>
      <Eui.EuiModalBody>
        <Eui.EuiBasicTable
          items={integrationDataTableData}
          columns={INTEGRATION_DATA_TABLE_COLUMNS}
        />
        <Eui.EuiSpacer />
        <Eui.EuiButton onClick={close} size="s">
          Close
        </Eui.EuiButton>
      </Eui.EuiModalBody>
    </Eui.EuiModal>
  );
}

export function SetupIntegrationForm({ config, updateConfig }: IntegrationConfigProps) {
  const connectionType =
    INTEGRATION_CONNECTION_DATA_SOURCE_TYPES.get(config.connectionType) ?? 'index';
  const indefiniteArticle = 'aeiou'.includes(connectionType.charAt(0)) ? 'an' : 'a';
  const capitalizedConnectionType =
    connectionType.charAt(0).toUpperCase() + connectionType.slice(1);

  const [dataSourceSuggestions, setDataSourceSuggestions] = useState(
    [] as Array<{ label: string }>
  );
  const [isSuggestionsLoading, setIsSuggestionsLoading] = useState(true);
  useEffect(() => {
    const updateDataSources = async () => {
      const data = await suggestDataSources(config.connectionType);
      setDataSourceSuggestions(data);
      setIsSuggestionsLoading(false);
    };

    setIsSuggestionsLoading(true);
    updateDataSources();
  }, [config.connectionType]);

  return (
    <Eui.EuiForm>
      <Eui.EuiTitle>
        <h1>Set Up Integration</h1>
      </Eui.EuiTitle>
      <Eui.EuiSpacer />
      <Eui.EuiTitle>
        <h2>Integration Details</h2>
      </Eui.EuiTitle>
      <Eui.EuiSpacer />
      <Eui.EuiFormRow label="Display Name">
        <Eui.EuiFieldText
          value={config.displayName}
          onChange={(event) => updateConfig({ displayName: event.target.value })}
        />
      </Eui.EuiFormRow>
      <Eui.EuiSpacer />
      <Eui.EuiTitle>
        <h2>Integration Connection</h2>
      </Eui.EuiTitle>
      <Eui.EuiSpacer />
      <Eui.EuiFormRow label="Data Source" helpText="Select a data source to connect to.">
        <Eui.EuiSelect
          options={integrationConnectionSelectorItems}
          value={config.connectionType}
          onChange={(event) =>
            updateConfig({ connectionType: event.target.value, connectionDataSource: '' })
          }
        />
      </Eui.EuiFormRow>
      <Eui.EuiFormRow
        label={capitalizedConnectionType}
        helpText={`Select ${indefiniteArticle} ${connectionType} to pull the data from.`}
      >
        <Eui.EuiComboBox
          options={dataSourceSuggestions}
          isLoading={isSuggestionsLoading}
          onChange={(selected) => {
            if (selected.length === 0) {
              updateConfig({ connectionDataSource: '' });
            } else {
              updateConfig({ connectionDataSource: selected[0].label });
            }
          }}
          selectedOptions={[{ label: config.connectionDataSource }]}
          singleSelection={{ asPlainText: true }}
        />
      </Eui.EuiFormRow>
      <Eui.EuiButton>Validate</Eui.EuiButton>
    </Eui.EuiForm>
  );
}

export function SetupBottomBar({ config }: { config: IntegrationConfig }) {
  return (
    <Eui.EuiBottomBar>
      <Eui.EuiFlexGroup justifyContent="flexEnd">
        <Eui.EuiFlexItem grow={false}>
          <Eui.EuiButton
            color="secondary"
            iconType="cross"
            onClick={() => {
              // TODO evil hack because props aren't set up
              let hash = window.location.hash;
              hash = hash.trim();
              hash = hash.substring(0, hash.lastIndexOf('/setup'));
              window.location.hash = hash;
            }}
          >
            Discard
          </Eui.EuiButton>
        </Eui.EuiFlexItem>
        <Eui.EuiFlexItem grow={false}>
          <Eui.EuiButton
            fill
            iconType="arrowRight"
            iconSide="right"
            onClick={() => console.log(config)}
          >
            Add Integration
          </Eui.EuiButton>
        </Eui.EuiFlexItem>
      </Eui.EuiFlexGroup>
    </Eui.EuiBottomBar>
  );
}

export function SetupIntegrationPage({ integration }: { integration: string }) {
  const [integConfig, setConfig] = useState({
    displayName: `${integration} Integration`,
    connectionType: 'index',
    connectionDataSource: '',
  } as IntegrationConfig);

  const updateConfig = (updates: Partial<IntegrationConfig>) =>
    setConfig(Object.assign({}, integConfig, updates));

  return (
    <Eui.EuiPage>
      <Eui.EuiPageBody>
        <Eui.EuiPageContent>
          <Eui.EuiPageContentBody restrictWidth>
            <SetupIntegrationForm config={integConfig} updateConfig={updateConfig} />
          </Eui.EuiPageContentBody>
        </Eui.EuiPageContent>
        <SetupBottomBar config={integConfig} />
      </Eui.EuiPageBody>
    </Eui.EuiPage>
  );
}
