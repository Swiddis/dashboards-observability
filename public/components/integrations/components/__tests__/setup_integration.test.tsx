/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { SetupIntegrationStepsPage } from '../setup_integration';

describe('Integration Setup Page Test', () => {
  configure({ adapter: new Adapter() });

  it('Renders integration setup page as expected', async () => {
    const wrapper = mount(<SetupIntegrationStepsPage />);

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
