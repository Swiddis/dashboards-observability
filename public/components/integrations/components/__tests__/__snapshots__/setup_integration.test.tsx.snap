// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration Setup Page Test Renders integration setup page as expected 1`] = `
<SetupIntegrationStepsPage>
  <EuiPage>
    <div
      className="euiPage euiPage--paddingMedium euiPage--grow"
    >
      <EuiPageBody>
        <div
          className="euiPageBody euiPageBody--borderRadiusNone"
        >
          <EuiFlexGroup>
            <div
              className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <EuiFlexItem>
                <div
                  className="euiFlexItem"
                >
                  <EuiSteps
                    steps={
                      Array [
                        Object {
                          "children": <EuiText />,
                          "status": "",
                          "title": "Name Integration",
                        },
                        Object {
                          "children": <EuiText />,
                          "status": "disabled",
                          "title": "Select index or data source for integration",
                        },
                      ]
                    }
                  >
                    <div
                      className="euiSteps"
                    >
                      <EuiStep
                        headingElement="p"
                        key="0"
                        status=""
                        step={1}
                        title="Name Integration"
                      >
                        <div
                          className="euiStep"
                        >
                          <div
                            className="euiStep__titleWrapper"
                          >
                            <EuiStepNumber
                              className="euiStep__circle"
                              isHollow={false}
                              number={1}
                              status=""
                              titleSize="s"
                            >
                              <span
                                className="euiStepNumber euiStep__circle"
                              >
                                <EuiScreenReaderOnly>
                                  <span
                                    className="euiScreenReaderOnly"
                                  >
                                    Step 1
                                  </span>
                                </EuiScreenReaderOnly>
                                <span
                                  aria-hidden="true"
                                  className="euiStepNumber__number"
                                >
                                  1
                                </span>
                              </span>
                            </EuiStepNumber>
                            <EuiTitle
                              className="euiStep__title"
                              size="s"
                            >
                              <p
                                className="euiTitle euiTitle--small euiStep__title"
                              >
                                Name Integration
                              </p>
                            </EuiTitle>
                          </div>
                          <div
                            className="euiStep__content"
                          >
                            <EuiText>
                              <div
                                className="euiText euiText--medium"
                              />
                            </EuiText>
                          </div>
                        </div>
                      </EuiStep>
                      <EuiStep
                        headingElement="p"
                        key="1"
                        status="disabled"
                        step={2}
                        title="Select index or data source for integration"
                      >
                        <div
                          className="euiStep euiStep-isDisabled"
                        >
                          <div
                            className="euiStep__titleWrapper"
                          >
                            <EuiStepNumber
                              className="euiStep__circle"
                              isHollow={false}
                              number={2}
                              status="disabled"
                              titleSize="s"
                            >
                              <span
                                className="euiStepNumber euiStepNumber--disabled euiStep__circle"
                              >
                                <EuiScreenReaderOnly>
                                  <span
                                    className="euiScreenReaderOnly"
                                  >
                                    Step 2 is disabled
                                  </span>
                                </EuiScreenReaderOnly>
                                <span
                                  aria-hidden="true"
                                  className="euiStepNumber__number"
                                >
                                  2
                                </span>
                              </span>
                            </EuiStepNumber>
                            <EuiTitle
                              className="euiStep__title"
                              size="s"
                            >
                              <p
                                className="euiTitle euiTitle--small euiStep__title"
                              >
                                Select index or data source for integration
                              </p>
                            </EuiTitle>
                          </div>
                          <div
                            className="euiStep__content"
                          >
                            <EuiText>
                              <div
                                className="euiText euiText--medium"
                              />
                            </EuiText>
                          </div>
                        </div>
                      </EuiStep>
                    </div>
                  </EuiSteps>
                </div>
              </EuiFlexItem>
              <EuiFlexItem>
                <div
                  className="euiFlexItem"
                >
                  <SetupIntegrationStep
                    activeStep={0}
                    config={
                      Object {
                        "dataSourceDescription": "",
                        "dataSourceFileType": "parquet",
                        "dataSourceLocation": "",
                        "dataSourceName": "",
                        "existingDataSourceName": "",
                        "instanceName": "",
                        "useExisting": true,
                      }
                    }
                    updateConfig={[Function]}
                  >
                    <SetupIntegrationMetadata
                      name=""
                      setName={[Function]}
                    >
                      <EuiForm>
                        <div
                          className="euiForm"
                        >
                          <EuiTitle>
                            <h1
                              className="euiTitle euiTitle--medium"
                            >
                              Name Integration
                            </h1>
                          </EuiTitle>
                          <EuiFormRow
                            describedByIds={Array []}
                            display="row"
                            fullWidth={false}
                            hasChildLabel={true}
                            hasEmptyLabelSpace={false}
                            helpText="The name will be used to label the newly added integration"
                            label="Name"
                            labelType="label"
                          >
                            <div
                              className="euiFormRow"
                              id="random_html_id-row"
                            >
                              <div
                                className="euiFormRow__labelWrapper"
                              >
                                <EuiFormLabel
                                  className="euiFormRow__label"
                                  htmlFor="random_html_id"
                                  isFocused={false}
                                  type="label"
                                >
                                  <label
                                    className="euiFormLabel euiFormRow__label"
                                    htmlFor="random_html_id"
                                  >
                                    Name
                                  </label>
                                </EuiFormLabel>
                              </div>
                              <div
                                className="euiFormRow__fieldWrapper"
                              >
                                <EuiFieldText
                                  aria-describedby="random_html_id-help-0"
                                  id="random_html_id"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  value=""
                                >
                                  <EuiFormControlLayout
                                    fullWidth={false}
                                    inputId="random_html_id"
                                  >
                                    <div
                                      className="euiFormControlLayout"
                                    >
                                      <div
                                        className="euiFormControlLayout__childrenWrapper"
                                      >
                                        <EuiValidatableControl>
                                          <input
                                            aria-describedby="random_html_id-help-0"
                                            className="euiFieldText"
                                            id="random_html_id"
                                            onBlur={[Function]}
                                            onChange={[Function]}
                                            onFocus={[Function]}
                                            type="text"
                                            value=""
                                          />
                                        </EuiValidatableControl>
                                        <EuiFormControlLayoutIcons />
                                      </div>
                                    </div>
                                  </EuiFormControlLayout>
                                </EuiFieldText>
                                <EuiFormHelpText
                                  className="euiFormRow__text"
                                  id="random_html_id-help-0"
                                  key="The name will be used to label the newly added integration"
                                >
                                  <div
                                    className="euiFormHelpText euiFormRow__text"
                                    id="random_html_id-help-0"
                                  >
                                    The name will be used to label the newly added integration
                                  </div>
                                </EuiFormHelpText>
                              </div>
                            </div>
                          </EuiFormRow>
                        </div>
                      </EuiForm>
                    </SetupIntegrationMetadata>
                  </SetupIntegrationStep>
                </div>
              </EuiFlexItem>
            </div>
          </EuiFlexGroup>
          <SetupBottomBar
            config={
              Object {
                "dataSourceDescription": "",
                "dataSourceFileType": "parquet",
                "dataSourceLocation": "",
                "dataSourceName": "",
                "existingDataSourceName": "",
                "instanceName": "",
                "useExisting": true,
              }
            }
            setStep={[Function]}
            step={0}
          >
            <EuiBottomBar>
              <EuiPortal>
                <Portal
                  containerInfo={
                    <div>
                      <section
                        aria-label="Page level controls"
                        class="euiBottomBar euiBottomBar--fixed euiBottomBar--paddingMedium"
                      >
                        <h2
                          class="euiScreenReaderOnly"
                        >
                          Page level controls
                        </h2>
                        <div
                          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
                        >
                          <div
                            class="euiFlexItem euiFlexItem--flexGrowZero"
                          >
                            <button
                              class="euiButton euiButton--danger"
                              type="button"
                            >
                              <span
                                class="euiButtonContent euiButton__content"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                  focusable="false"
                                  height="16"
                                  role="img"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                />
                                <span
                                  class="euiButton__text"
                                >
                                  Cancel
                                </span>
                              </span>
                            </button>
                          </div>
                          <div
                            class="euiFlexItem"
                          >
                            <div
                              class="euiSpacer euiSpacer--l"
                            />
                          </div>
                          <div
                            class="euiFlexItem euiFlexItem--flexGrowZero"
                          >
                            <button
                              class="euiButton euiButton--primary euiButton--fill"
                              type="button"
                            >
                              <span
                                class="euiButtonContent euiButton__content"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                  focusable="false"
                                  height="16"
                                  role="img"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                />
                                <span
                                  class="euiButton__text"
                                >
                                  Next
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                      </section>
                      <p
                        aria-live="assertive"
                        class="euiScreenReaderOnly"
                      >
                        There is a new region landmark with page level controls at the end of the document.
                      </p>
                    </div>
                  }
                >
                  <EuiI18n
                    default="Page level controls"
                    token="euiBottomBar.screenReaderHeading"
                  >
                    <section
                      aria-label="Page level controls"
                      className="euiBottomBar euiBottomBar--fixed euiBottomBar--paddingMedium"
                      style={
                        Object {
                          "bottom": undefined,
                          "left": undefined,
                          "right": undefined,
                          "top": undefined,
                        }
                      }
                    >
                      <EuiScreenReaderOnly>
                        <h2
                          className="euiScreenReaderOnly"
                        >
                          Page level controls
                        </h2>
                      </EuiScreenReaderOnly>
                      <EuiFlexGroup
                        justifyContent="flexEnd"
                      >
                        <div
                          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
                        >
                          <EuiFlexItem
                            grow={false}
                          >
                            <div
                              className="euiFlexItem euiFlexItem--flexGrowZero"
                            >
                              <EuiButton
                                color="danger"
                                iconType="cross"
                                onClick={[Function]}
                              >
                                <EuiButtonDisplay
                                  baseClassName="euiButton"
                                  color="danger"
                                  disabled={false}
                                  element="button"
                                  iconType="cross"
                                  isDisabled={false}
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <button
                                    className="euiButton euiButton--danger"
                                    disabled={false}
                                    onClick={[Function]}
                                    style={
                                      Object {
                                        "minWidth": undefined,
                                      }
                                    }
                                    type="button"
                                  >
                                    <EuiButtonContent
                                      className="euiButton__content"
                                      iconSide="left"
                                      iconType="cross"
                                      textProps={
                                        Object {
                                          "className": "euiButton__text",
                                        }
                                      }
                                    >
                                      <span
                                        className="euiButtonContent euiButton__content"
                                      >
                                        <EuiIcon
                                          className="euiButtonContent__icon"
                                          color="inherit"
                                          size="m"
                                          type="cross"
                                        >
                                          <EuiIconEmpty
                                            aria-hidden={true}
                                            className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                            focusable="false"
                                            role="img"
                                            style={null}
                                          >
                                            <svg
                                              aria-hidden={true}
                                              className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                              focusable="false"
                                              height={16}
                                              role="img"
                                              style={null}
                                              viewBox="0 0 16 16"
                                              width={16}
                                              xmlns="http://www.w3.org/2000/svg"
                                            />
                                          </EuiIconEmpty>
                                        </EuiIcon>
                                        <span
                                          className="euiButton__text"
                                        >
                                          Cancel
                                        </span>
                                      </span>
                                    </EuiButtonContent>
                                  </button>
                                </EuiButtonDisplay>
                              </EuiButton>
                            </div>
                          </EuiFlexItem>
                          <EuiFlexItem>
                            <div
                              className="euiFlexItem"
                            >
                              <EuiSpacer>
                                <div
                                  className="euiSpacer euiSpacer--l"
                                />
                              </EuiSpacer>
                            </div>
                          </EuiFlexItem>
                          <EuiFlexItem
                            grow={false}
                          >
                            <div
                              className="euiFlexItem euiFlexItem--flexGrowZero"
                            >
                              <EuiButton
                                fill={true}
                                iconType="check"
                                onClick={[Function]}
                              >
                                <EuiButtonDisplay
                                  baseClassName="euiButton"
                                  disabled={false}
                                  element="button"
                                  fill={true}
                                  iconType="check"
                                  isDisabled={false}
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <button
                                    className="euiButton euiButton--primary euiButton--fill"
                                    disabled={false}
                                    onClick={[Function]}
                                    style={
                                      Object {
                                        "minWidth": undefined,
                                      }
                                    }
                                    type="button"
                                  >
                                    <EuiButtonContent
                                      className="euiButton__content"
                                      iconSide="left"
                                      iconType="check"
                                      textProps={
                                        Object {
                                          "className": "euiButton__text",
                                        }
                                      }
                                    >
                                      <span
                                        className="euiButtonContent euiButton__content"
                                      >
                                        <EuiIcon
                                          className="euiButtonContent__icon"
                                          color="inherit"
                                          size="m"
                                          type="check"
                                        >
                                          <EuiIconEmpty
                                            aria-hidden={true}
                                            className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                            focusable="false"
                                            role="img"
                                            style={null}
                                          >
                                            <svg
                                              aria-hidden={true}
                                              className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                              focusable="false"
                                              height={16}
                                              role="img"
                                              style={null}
                                              viewBox="0 0 16 16"
                                              width={16}
                                              xmlns="http://www.w3.org/2000/svg"
                                            />
                                          </EuiIconEmpty>
                                        </EuiIcon>
                                        <span
                                          className="euiButton__text"
                                        >
                                          Next
                                        </span>
                                      </span>
                                    </EuiButtonContent>
                                  </button>
                                </EuiButtonDisplay>
                              </EuiButton>
                            </div>
                          </EuiFlexItem>
                        </div>
                      </EuiFlexGroup>
                    </section>
                  </EuiI18n>
                  <EuiScreenReaderOnly>
                    <p
                      aria-live="assertive"
                      className="euiScreenReaderOnly"
                    >
                      <EuiI18n
                        default="There is a new region landmark with page level controls at the end of the document."
                        token="euiBottomBar.screenReaderAnnouncement"
                      >
                        There is a new region landmark with page level controls at the end of the document.
                      </EuiI18n>
                    </p>
                  </EuiScreenReaderOnly>
                </Portal>
              </EuiPortal>
            </EuiBottomBar>
          </SetupBottomBar>
        </div>
      </EuiPageBody>
    </div>
  </EuiPage>
</SetupIntegrationStepsPage>
`;
